mam/v2/api/api.c

```diff
@@ -166,12 +168,60 @@ retcode_t mam_api_bundle_write_packet(mam_api_t *const api,
   return RC_OK;
 }
 
-retcode_t mam_api_bundle_read(mam_api_t *const api,
-                              mam_channel_t const *const cha,
-                              bundle_transactions_t const *const bundle,
-                              flex_trit_t *const payload) {
-  if (api == NULL || cha == NULL || bundle == NULL || payload == NULL) {
-    return RC_NULL_PARAM;
+bool mam_api_bundle_contains_header(bundle_transactions_t const *const bundle) {
+  iota_transaction_t *tx = (iota_transaction_t *)utarray_front(bundle);
+  return flex_trits_are_null(transaction_tag(tx), NUM_TRITS_TAG);
+}
+
+retcode_t mam_api_bundle_read_msg(mam_recv_msg_context_t *const ctx,
```

Comment:
I think we should have `mam_api_bundle_read_header` and `mam_api_bundle_read_packet`. And `mam_api_bundle_read_msg` would call them depending on if it finds a header or not. Otherwise we'll have duplicate code in `mam_api_bundle_read_packet`.

Reply:
yes